src/OpenTelemetry/.publicApi/net462/PublicAPI.Unshipped.txt

```diff
@@ -1,4 +1,49 @@
 #nullable enable
+OpenTelemetry.Logs.LogEmitter
```

Comment:
Here is my suggestion:
1. Use this PR to focus on the LogRecord improvements (including polling, ref count, setters) - this would give good perf benefits, and set a cornerstone for LogEmitter or any bridge with Serilog/NLog/EventSource/etc.
2. LogEmitter can either be a separate PR or included here, as long as the scope of the change is not too big + is self-contained (e.g. if LogEmitter is need to make meaningful perf tests, probably fine to keep it, but it has to be kept `internal`)
3. Before the spec becomes stable, we can either keep LogEmitter `internal` (for our own testing purpose), or making it public AND having it in a separate package (similar to OTLP Log Exporter) - if we know someone who is interested in trying it out.

Reply:
Sounds good. 👍 to making `LogEmitter` internal and leaving it where it is in this PR.

Reply:
* Let's make it internal and keep where it is. We can figure out where to move it if needed before it goes public?

* What shall I do with `LogRecordData` + `LogRecordAttributeList`?
  * If `LogRecordData` goes internal, so to will `LogRecord.GetDataRef()` which gives exporters a way to get at the data more quickly and also gives users a way to mutate _everything_ on `LogRecord` (which has been requested).
  * If `LogRecordAttributeList` goes internal, no real downside without `LogEmitter`. I do have this in there...
    https://github.com/open-telemetry/opentelemetry-dotnet/blob/2114dcaae6026b4927f5e38360808436329d1f83/src/OpenTelemetry/Logs/OpenTelemetryLogger.cs#L93

    What that allows you to do is pass `LogRecordAttributeList` as `TState` to `ILogger.Log<TState>` and get boxing/allocation-free tags. If it goes internal, users won't be able to easily build upon that perf-path.